homeassistant/components/google_assistant/trait.py

```diff
@@ -161,7 +161,7 @@
 COMMAND_LOCATE = f"{PREFIX_COMMANDS}Locate"
 COMMAND_CHARGE = f"{PREFIX_COMMANDS}Charge"
 
-TRAITS = []
+TRAITS: list[type[_Trait]] = []
 
```

Comment:
Is this some abstract traits type? Im not familiar with this typing syntax?

Reply:
Added type annotation to `register_trait` as well to make it a bit clearer.

`register_trait` is used as a class decorator to add classes (not instances!) to the `TRAITS` list. All those classes inherit from `_Trait` and since it's the classes themselves and not instances, I needed to add `type[...]`.

https://github.com/home-assistant/core/blob/0530f61373395db83b016173399b06eb8337a39a/homeassistant/components/google_assistant/trait.py#L281-L283

Reply:
I think i was mainly confused by the _Trait name. Didnt realise the class was prefixed by _ to indicate private (for no apparent reason).